pulser-core/pulser/waveforms.py

```diff
@@ -734,13 +772,18 @@ def _to_abstract_repr(self) -> dict[str, Any]:
         return abstract_repr("BlackmanWaveform", self._duration, self._area)
 
     def __str__(self) -> str:
-        return f"Blackman(Area: {self._area:.3g})"
+        return f"Blackman(Area: {float(self._area):.3g})"
 
     def __repr__(self) -> str:
-        return f"BlackmanWaveform({self._duration} ns, Area: {self._area:.3g})"
+        return (
+            f"BlackmanWaveform({self._duration} ns, "
+            f"Area: {float(self._area):.3g})"
+        )
 
-    def __mul__(self, other: float) -> BlackmanWaveform:
-        return BlackmanWaveform(self._duration, self._area * float(other))
+    def __mul__(self, other: float | ArrayLike) -> BlackmanWaveform:
+        return BlackmanWaveform(
+            self._duration, self._area * pm.AbstractArray(other, dtype=float)
+        )
 
 
 class InterpolatedWaveform(Waveform):
```

Comment:
We don't touch the type of values in InterpolatedWaveform because we don't have the interpolator in torch ?

Reply:
Yeah, pretty much